crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -66,23 +66,43 @@ pub struct DeferredPrepass;
 pub struct ViewPrepassTextures {
     /// The depth texture generated by the prepass.
     /// Exists only if [`DepthPrepass`] is added to the [`ViewTarget`](bevy_render::view::ViewTarget)
-    pub depth: Option<CachedTexture>,
+    pub depth: Option<ColorAttachment>,
     /// The normals texture generated by the prepass.
     /// Exists only if [`NormalPrepass`] is added to the [`ViewTarget`](bevy_render::view::ViewTarget)
-    pub normal: Option<CachedTexture>,
+    pub normal: Option<ColorAttachment>,
     /// The motion vectors texture generated by the prepass.
     /// Exists only if [`MotionVectorPrepass`] is added to the `ViewTarget`
-    pub motion_vectors: Option<CachedTexture>,
+    pub motion_vectors: Option<ColorAttachment>,
     /// The deferred gbuffer generated by the deferred pass.
     /// Exists only if [`DeferredPrepass`] is added to the `ViewTarget`
-    pub deferred: Option<CachedTexture>,
+    pub deferred: Option<ColorAttachment>,
     /// A texture that specifies the deferred lighting pass id for a material.
     /// Exists only if [`DeferredPrepass`] is added to the `ViewTarget`
     pub deferred_lighting_pass_id: Option<CachedTexture>,
     /// The size of the textures.
     pub size: Extent3d,
 }
 
+impl ViewPrepassTextures {
```

Comment:
Instead of having a separate function for each texture, I think `ColorAttachment` could have a `.view()` that gets the default_view of the `CachedTexture`

Reply:
We need these functions anyways, because you need to do some mapping of Option<T> to Option<&U>, and the caller code gets messy pretty quickly. I'd rather keep it as-is.